content/docs/command-reference/pull.md

```diff
@@ -107,89 +107,146 @@ reflinks or hardlinks to put it in the workspace without copying. See
+├── ...
 ```
 
-In this case we left off the `--remote` option, so it will have pulled from the
-default remote. The only files considered in this case are what is listed in the
-`out` field of the DVC-file `targets`.
+We can download specific <abbr>outputs</abbr> of a single DVC-file:
+
+```dvc
+$ dvc pull prepare.dvc
+```
 
 ## Example: With dependencies
 
-Demonstrating the `--with-deps` option requires a larger example. First, assume
-a [pipeline](/doc/command-reference/pipeline) has been setup with these
+Our [pipeline](/doc/command-reference/pipeline) has been setup with these
 [stages](/doc/command-reference/run):
 
 ```dvc
-$ dvc pipeline show
-
-data/Posts.xml.zip.dvc
-Posts.xml.dvc
-Posts.tsv.dvc
-Posts-test.tsv.dvc
-matrix-train.p.dvc
-model.p.dvc
-Dvcfile
+$ dvc pipeline show evaluate.dvc
+data/data.xml.dvc
+prepare.dvc
+featurize.dvc
+train.dvc
+evaluate.dvc
 ```
 
 Imagine the remote storage has been modified such that the data in some of these
 stages should be updated in the <abbr>workspace</abbr>.
 
 ```dvc
-$ dvc status --cloud
-
-  deleted:            data/model.p
-  deleted:            data/matrix-test.p
-  deleted:            data/matrix-train.p
+$ dvc status -c
+    deleted:            data/features/test.pkl
+    deleted:            data/features/train.pkl
+    deleted:            model.pkl
+    deleted:            data/prepared/test.tsv
+    deleted:            data/prepared/train.tsv
+    deleted:            data/data.xml
 ```
 
 One could do a simple `dvc pull` to get all the data, but what if you only want
 to retrieve part of the data?
 
 ```dvc
-$ dvc pull --remote r1 --with-deps matrix-train.p.dvc
+$ dvc pull --with-deps featurize.dvc
 
 ... Do some work based on the partial update
 
-$ dvc pull --remote r1 --with-deps model.p.dvc
+$ dvc pull --with-deps evaluate.dvc
 
 ... Pull the rest of the data
 
-$ dvc pull --remote r1
+$ dvc pull
 
 Everything is up to date.
 ```
 
 With the first `dvc pull` we specified a stage in the middle of this pipeline
-(`matrix-train.p.dvc`) while using `--with-deps`. DVC started with that DVC-file
-and searched backwards through the pipeline for data files to download. Because
-the `model.p.dvc` stage occurs later, its data was not pulled.
+(`featurize.dvc`) while using `--with-deps`. DVC started with that DVC-file and
+searched backwards through the pipeline for data files to download. Because the
+`evaluate.dvc` stage occurs later, its data was not pulled.
 
-Then we ran `dvc pull` specifying the last stage, `model.p.dvc`, and its data
+Then we ran `dvc pull` specifying the last stage, `evaluate.dvc`, and its data
 was downloaded. Finally, we ran `dvc pull` with no flags to make sure that all
 data was already pulled with the previous commands.
+
+> Note that if you followed previous example, then `dvc status -c` would output
+> `Data and pipelines are up to date.` The last stage file `evaluate.dvc`
+> doesn't add any more data files than those form previous stages
+
```

Comment:
Hmmm. I see why this note is there bu we shouldn't need it. Users may indeed follow examples one by one and that would kind of invalidate this whole example — it would fail before they get to this note. It would be best to add a step to make sure the data in question is deleted for the example. Maybe as a block quote (like this note) or in an expandable `<details>` section like in the 1st example has. Or if you can think of a better way, please suggest.

Reply:
Also a couple typos in there 😋 but np since we'll be removing this text.

Reply:
Just this is missing it seems:

> It would be best to add a step to make sure the data in question is deleted for the example.